pkg/controller/machineset/controller_test.go

```diff
@@ -262,3 +266,76 @@ func TestAdoptOrphan(t *testing.T) {
 		}
 	}
 }
+
+var _ = Describe("MachineSet Reconcile", func() {
+	var r *ReconcileMachineSet
+	var result reconcile.Result
+	var reconcileErr error
+	var rec *record.FakeRecorder
+
+	BeforeEach(func() {
+		Expect(v1beta1.AddToScheme(scheme.Scheme)).To(Succeed())
+		rec = record.NewFakeRecorder(32)
+
+		r = &ReconcileMachineSet{
+			scheme:   scheme.Scheme,
+			recorder: rec,
+		}
+	})
+
+	JustBeforeEach(func() {
+		request := reconcile.Request{NamespacedName: types.NamespacedName{Name: "machineset1", Namespace: "default"}}
+		result, reconcileErr = r.Reconcile(request)
+	})
+
+	Context("with a valid machineset", func() {
+		BeforeEach(func() {
+			dt := metav1.Now()
+
+			ms := &v1beta1.MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "machineset1",
+					Namespace:         "default",
+					DeletionTimestamp: &dt,
+				},
+				Spec: v1beta1.MachineSetSpec{
+					Template: v1beta1.MachineTemplateSpec{},
+				}}
+
+			r.Client = fake.NewFakeClientWithScheme(scheme.Scheme, ms)
+		})
+
+		It("has the expected result", func() {
+			Expect(result).To(Equal(reconcile.Result{}))
+		})
+
+		It("has the expected error", func() {
+			Expect(reconcileErr).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("with an invalid machineset", func() {
+		BeforeEach(func() {
+			var replicas int32
+			ms := &v1beta1.MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machineset1",
+					Namespace: "default",
+				},
+				Spec: v1beta1.MachineSetSpec{
+					Replicas: &replicas,
+				},
+			}
+
+			ms.Spec.Selector.MatchLabels = map[string]string{
+				"--$-invalid": "true",
+			}
+
+			r.Client = fake.NewFakeClientWithScheme(scheme.Scheme, ms)
+		})
+
+		It("did something with events", func() {
+			Eventually(rec.Events).Should(Receive())
+		})
```

Comment:
Is it possible to also check the result and error here? Do you know what they're expected to be? /could you work it out from the code? (Could be a follow up)

Reply:
That should be possible with `Eventually(rec.Events).Should(Receive(Equal("..")))`